### ISSUE TITLE
API suggestion: connected_components_labeling

### ISSUE THREAD

**Issue by jni** [clEsperanto/pyclesperanto_prototype#61]
Currently, different versions of connected_components_labeling are spread out over different functions, e.g. `connected_components_labeling_box`. The API would be simpler if it were a single function with a connectivity parameter. [Quoting](https://github.com/clEsperanto/pyclesperanto_prototype/pull/48/files#r531216268) @haesleinhuepf:

> Box: 8- resp. 26-connected (3d/2d), partly separable
> Diamond: 4- resp. 6-connected, not separable
>
> I engineered this terminology very carefully. And I'd love to know how to improve it.

And [my response](https://github.com/clEsperanto/pyclesperanto_prototype/pull/48/files#r547758701) (only a month late 😅):

> I prefer to use `connected_components_labeling(binary, connectivity={1...ndim})` (see [`ndimage.label`](https://docs.scipy.org/doc/scipy/reference/generated/scipy.ndimage.label.html)/[`ndimage.generate_binary_structure`](https://docs.scipy.org/doc/scipy/reference/generated/scipy.ndimage.generate_binary_structure.html#scipy.ndimage.generate_binary_structure)/[`skimage.measure.label`](https://scikit-image.org/docs/dev/api/skimage.measure.html#skimage.measure.label)). Separability is an implementation detail that you can handle internally (when connectivity==ndim).

**Comment by jni** [comment]
It might even be worth calling it just "label" to match the above-linked functions, but probably not — I always thought that name was not descriptive enough!

**Comment by haesleinhuepf** [comment]
@jni 
I like the idea! We could do this for all box/sphere filters, and have box as default (because it's faster). That would make thing easier to read 🙂

Regarding naming: How about this scheme?
* label_spots (single pixels)
* label_connected_components (regions)
* label_voronoi (the whole image, partioned)

**Comment by jni** [comment]
Nice! Is `label_spots` [this function](https://github.com/scikit-image/scikit-image/issues/5087)? 😃

**Comment by haesleinhuepf** [comment]
> Nice! Is `label_spots` [this function](https://github.com/scikit-image/scikit-image/issues/5087)? 😃

No, `label_spots` [takes a binary image to label pixels](https://clij.github.io/clij2-docs/reference_labelSpots). It assumes that only single isolated pixels are 1. It's a faster version of `connected_component_labeling`.

The method you look for is called [pointlist_to_labelled_spots](https://clij.github.io/clij2-docs/reference_pointlistToLabelledSpots)

Btw. discussions about optimal naming in the API are appreciated! 🙏
